src/HttpCache/EventListener/AddTagsListener.php

```diff
@@ -25,16 +27,25 @@
  * @see https://support.cloudflare.com/hc/en-us/articles/206596608-How-to-Purge-Cache-Using-Cache-Tags-Enterprise-only-
  *
  * @author Kévin Dunglas <dunglas@gmail.com>
+ * @author Yanick Witschi <yanick.witschi@terminal42.ch>
  *
  * @experimental
  */
 final class AddTagsListener
 {
     private $iriConverter;
+    private $tagsFormatter;
 
-    public function __construct(IriConverterInterface $iriConverter)
+    public function __construct(IriConverterInterface $iriConverter, CacheTagsFormatterInterface $tagsFormatter = null)
     {
         $this->iriConverter = $iriConverter;
+        $this->tagsFormatter = $tagsFormatter;
+
+        // BC
+        if (null === $this->tagsFormatter) {
+            @trigger_error('Passing no implementation of the CacheTagsFormatterInterface is deprecated since version 2.3 and will be removed in 3.0.', E_USER_DEPRECATED);
```

Comment:
`Passing an object that does not implement CacheTagsFormatterInterface is deprecated since version 2.3 and will be removed in 3.0.` WDYT ?